crates/trigger/src/cli.rs

```diff
@@ -56,6 +60,47 @@ where
     )]
     pub cache: Option<PathBuf>,
 
+    /// Disable Wasmtime's pooling instance allocator.
+    #[cfg(not(target_os = "windows"))]
+    #[clap(long = "disable-pooling")]
+    pub disable_pooling: bool,
```

Comment:
Really appreciate that this is configurable on the command line. Do you have any considerations re: this being opt-in vs. opt-out by default? 

Specifically, should one think about this as an extra optimization to enable when needed, or is it something we should encourage everyone to always use?

Reply:
That's a good question. Given the pretty massive wins this brings, I'd argue it should be enabled by default. However, this isn't an entirely clear-cut choice, because it introduces a new failure mode, where components use more resources than are allocated for each instance in the pooling allocator configuration.

Additionally, the pooling allocator really only makes sense for use cases with short component lifetimes. I think that should apply to Spin pretty generally though?